server/src/main/java/org/opensearch/common/xcontent/support/XContentHttpChunk.java

```diff
@@ -42,6 +47,13 @@ private XContentHttpChunk(@Nullable final XContentBuilder builder) {
         if (builder == null /* no content */) {
             content = BytesArray.EMPTY;
         } else {
+            // Always finalize the output chunk with '\r\n' sequence
+            try (final OutputStream out = builder.getOutputStream()) {
+                builder.close();
+                out.write(CHUNK_SEPARATOR);
```

Comment:
This seems like we're going around the XContentBuilder to write directly to an output stream. The javadoc actually says this is ["dangerous"](https://github.com/opensearch-project/OpenSearch/blob/bf4367877eab27dff05a74d683d14e820130172d/libs/core/src/main/java/org/opensearch/core/xcontent/XContentBuilder.java#L241). Could we instead just do the following?

```java
return CompositeBytesReference.of(BytesReference.bytes(builder), new BytesArray(CHUNK_SEPARATOR));
```

Reply:
> This seems like we're going around the XContentBuilder to write directly to an output stream.

Correct, this is why we close the stream before writing, but `CompositeBytesReference` looks less "hacky", let me try it, thanks!